velox/exec/tests/ArrowStreamTest.cpp

```diff
@@ -0,0 +1,241 @@
+    }
+  };
+
+  /// A mock Arrow reader returns an Arrow array for each call of GetNext.
+  class ArrowReader {
+   public:
+    ArrowReader(
+        std::shared_ptr<memory::MemoryPool> pool,
+        const std::vector<RowVectorPtr>& vectors,
+        const TypePtr& type)
+        : pool_(pool), vectors_(vectors), type_(type) {}
+
+    int GetNext(struct ArrowArray* outArray) {
+      if (vectorIndex_ < vectors_.size()) {
+        exportToArrow(vectors_[vectorIndex_], *outArray, pool_.get());
+        vectorIndex_ += 1;
+        Helper::ArrowArrayMockRelease(outArray);
+      } else {
+        // End of stream.
+        Helper::ArrowArrayMarkReleased(outArray);
+      }
+      return 0;
+    }
+
+    void GetArrowSchema(ArrowSchema& out) {
+      exportToArrow(BaseVector::create(type_, 0, pool_.get()), out);
+      out.release = Helper::mockSchemaRelease;
+    }
+
+   private:
+    std::shared_ptr<memory::MemoryPool> pool_;
+    std::vector<RowVectorPtr> vectors_;
+    TypePtr type_;
+    int vectorIndex_ = 0;
+  };
+
+  /// For the export from Arrow reader to Arrow array stream.
+  class ExportedArrayStream {
+   public:
+    struct PrivateData {
+      explicit PrivateData(std::shared_ptr<ArrowReader> reader)
+          : reader_(std::move(reader)) {}
+
+      std::shared_ptr<ArrowReader> reader_;
+      std::string last_error_;
+
+      PrivateData() = default;
+      DISALLOW_COPY_AND_ASSIGN(PrivateData);
+    };
+
+    explicit ExportedArrayStream(struct ArrowArrayStream* stream)
+        : stream_(stream) {}
+
+    int GetSchema(struct ArrowSchema* outSchema) {
+      reader()->GetArrowSchema(*outSchema);
+      return 0;
+    }
+
+    int GetNext(struct ArrowArray* outArray) {
+      return reader()->GetNext(outArray);
+    }
+
+    const char* GetLastError() {
+      const auto& lastError = private_data()->last_error_;
+      return lastError.empty() ? nullptr : lastError.c_str();
+    }
+
+    void Release() {
+      if (Helper::ArrowArrayStreamIsReleased(stream_)) {
+        return;
+      }
+      DCHECK_NE(private_data(), nullptr);
+      delete private_data();
+
+      Helper::ArrowArrayStreamMarkReleased(stream_);
+    }
+
+    static int StaticGetSchema(
+        struct ArrowArrayStream* stream,
+        struct ArrowSchema* outSchema) {
+      ExportedArrayStream self{stream};
+      return self.GetSchema(outSchema);
+    }
+
+    static int StaticGetNext(
+        struct ArrowArrayStream* stream,
+        struct ArrowArray* outArray) {
+      ExportedArrayStream self{stream};
+      return self.GetNext(outArray);
+    }
+
+    static void StaticRelease(struct ArrowArrayStream* stream) {
+      ExportedArrayStream{stream}.Release();
+    }
+
+    static const char* StaticGetLastError(struct ArrowArrayStream* stream) {
+      return ExportedArrayStream{stream}.GetLastError();
+    }
+
+   private:
+    std::shared_ptr<memory::MemoryPool> pool_;
+    struct ArrowArrayStream* stream_;
+    PrivateData* private_data() {
```

Comment:
naming: privateData()

Move methods before the member variables.

Reply:
Fixed.